src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -110,34 +110,22 @@ public virtual bool TryCreate(FunctionMetadata functionMetadata, out FunctionDes
         protected virtual ParameterDescriptor CreateTriggerParameter(BindingMetadata triggerMetadata, Type parameterType = null)
         {
             ParameterDescriptor triggerParameter = null;
-            switch (triggerMetadata.Type)
+            string type = triggerMetadata.Type.ToLowerInvariant();
+            switch (type)
             {
-                case BindingType.QueueTrigger:
-                    parameterType = DefaultParameterType(parameterType, triggerMetadata.DataType);
-                    triggerParameter = ParseQueueTrigger((QueueBindingMetadata)triggerMetadata, parameterType);
-                    break;
-                case BindingType.EventHubTrigger:
-                    parameterType = DefaultParameterType(parameterType, triggerMetadata.DataType);
-                    triggerParameter = ParseEventHubTrigger((EventHubBindingMetadata)triggerMetadata, parameterType);
-                    break;
-                case BindingType.BlobTrigger:
-                    triggerParameter = ParseBlobTrigger((BlobBindingMetadata)triggerMetadata, parameterType ?? typeof(Stream));
-                    break;
-                case BindingType.ServiceBusTrigger:
-                    parameterType = DefaultParameterType(parameterType, triggerMetadata.DataType);
-                    triggerParameter = ParseServiceBusTrigger((ServiceBusBindingMetadata)triggerMetadata, parameterType);
-                    break;
-                case BindingType.TimerTrigger:
-                    triggerParameter = ParseTimerTrigger((TimerBindingMetadata)triggerMetadata, parameterType ?? typeof(TimerInfo));
-                    break;
-                case BindingType.HttpTrigger:
+                case "httptrigger":
                     triggerParameter = ParseHttpTrigger((HttpTriggerBindingMetadata)triggerMetadata, parameterType ?? typeof(HttpRequestMessage));
                     break;
-                case BindingType.ManualTrigger:
+                case "manualtrigger":
                     triggerParameter = ParseManualTrigger(triggerMetadata, parameterType ?? typeof(string));
                     break;
-                case BindingType.ApiHubFileTrigger:
-                    triggerParameter = ParseApiHubFileTrigger((ApiHubBindingMetadata)triggerMetadata, parameterType ?? typeof(Stream));
+                default:
+                    if (triggerMetadata.Raw == null)
+                    {
+                        // TEMP: This conversion is only here to keep unit tests passing
+                        triggerMetadata.Raw = JObject.FromObject(triggerMetadata);
+                    }
```

Comment:
When onboarding an extension, you previously had to add a switch above - this is now all dynamic based on the providers.